auth/api/iam/api.go

```diff
@@ -143,6 +144,60 @@ func (r Wrapper) HandleTokenRequest(ctx context.Context, request HandleTokenRequ
 	}
 }
 
+// IntrospectAccessToken allows the resource server (XIS/EHR) to introspect details of an access token issued by this node
+func (r Wrapper) IntrospectAccessToken(ctx context.Context, request IntrospectAccessTokenRequestObject) (IntrospectAccessTokenResponseObject, error) {
+	// Client authentication if required, return 401 if fails.
+	// TODO: Is this relevant? Should be handled by API authentication middleware
+	clientAuthorized := true
+	if !clientAuthorized {
+		return IntrospectAccessToken401Response{}, nil
+	}
+
+	// Validate token
+	if request.Body.Token == "" {
+		// Return 200 + 'Active = false' when token is invalid or malformed
+		return IntrospectAccessToken200JSONResponse{}, nil
+	}
+
+	token := AccessToken{}
+	if err := r.s2sAccessTokenStore().Get(request.Body.Token, &token); err != nil {
+		// Return 200 + 'Active = false' when token is invalid or malformed
+		return IntrospectAccessToken200JSONResponse{}, err
+	}
+
+	if token.Expiration.Before(time.Now()) {
+		// Return 200 + 'Active = false' when token is invalid or malformed
+		// can happen between token expiration and pruning of database
+		return IntrospectAccessToken200JSONResponse{}, nil
+	}
+
```

Comment:
should we check `not before`?

Reply:
Edit: I see we don't set it.

Reply:
We require `exp - iat <= 5s`, so further reducing that by adding `nbf` in-between doesn't make sense to me.